examples/ecdsa_utils/ecdsa_utils.h

```diff
@@ -37,31 +37,62 @@
 extern "C" {
 #endif
 
-/**
- * @brief decodes two ASN.1 integers to the R and S components of a ECC signature
- * @param[in] asn1 buffer containing the ASN.1 encoded R and S values
- * @param[in] asn1_len length of the asn1 buffer
- * @param[out] rs buffer where to write the R and S values
- * @param[in,out] rs_len length of the rs buffer, contains the bytes written afterwards
- * @returns OPTIGA_LIB_SUCCESS on success, OPTIGA_LIB_ERROR else
- */
-optiga_lib_status_t asn1_to_ecdsa_rs(const uint8_t * asn1, size_t asn1_len,
-									 uint8_t * rs, size_t * rs_len);
+#include <stdbool.h>
+#include <stdint.h>
+#include <stddef.h>
+
+// Encode two integers in DER format
+// TAG + LENGTH needs 2 bytes
+// if the highest bit of the integer is set we need an extra byte
+/** @brief Maximum overhead of the ASN.1 encoding for the R and S components of an ECDSA signature */
+#define ECDSA_RS_MAX_ASN1_OVERHEAD ((2 + 1) * 2)
 
 /**
- * @brief Encodes the ECDSA signature components (r, s) in ASN.1 format.
+ * @brief Encodes the ECDSA signature components (r, s) in as two ASN.1 INTEGERs concatenated. This is the format

```

Comment:
A bit better: "Encodes the ECDSA signature components (r, s) as two concatenated ASN.1 INTEGERs."